### PR TITLE
[AppConfig]Add customer managed key when updating stores. 

### PR DESCRIPTION
**Pull request by shenmuxiaosen**
**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S165.

File: src/azure-cli/azure/cli/command_modules/appconfig/_params.py [hunk 1]
```diff
@@ -79,6 +79,12 @@ def load_arguments(self, _):
     with self.argument_context('appconfig update') as c:
         c.argument('tags', arg_type=tags_type)
 
+    with self.argument_context('appconfig update', arg_group='Customer Managed Key', is_preview=True) as c:
+        c.argument('encryption_key_name', help='The name of the KeyVault key.')
+        c.argument('encryption_key_vault', help='The Uri of the KeyVault.')
+        c.argument('encryption_key_version', help='The version of the KeyVault key. Use the latest version by default.')
+        c.argument('identity_client_id', help='Client id of the managed identity with wrap and unwrap access to encryption key. Use system assigned identity by default.')
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appconfig/_params.py]: ID

File: src/azure-cli/azure/cli/command_modules/appconfig/_params.py [hunk 2]
```diff
@@ -79,6 +79,12 @@ def load_arguments(self, _):
     with self.argument_context('appconfig update') as c:
         c.argument('tags', arg_type=tags_type)
 
+    with self.argument_context('appconfig update', arg_group='Customer Managed Key', is_preview=True) as c:
+        c.argument('encryption_key_name', help='The name of the KeyVault key.')
+        c.argument('encryption_key_vault', help='The Uri of the KeyVault.')
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appconfig/_params.py]: URI

File: src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py [hunk 3]
```diff
@@ -1111,25 +1078,6 @@ def test_azconfig_feature(self, resource_group, location):
...
-                                   self.check('[0].description', updated_entry_description),
-                                   self.check('[0].label', entry_label),
-                                   self.check('[0].state', default_state),
-                                   self.check('[1].locked', default_locked),
-                                   self.check('[1].key', suffix_feature),
-                                   self.check('[1].description', default_description),
-                                   self.check('[1].label', null_label),
-                                   self.check('[1].state', default_state)]).get_output_in_json()
-        assert len(deleted) == 2
-
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py]: Why you delete these testing code

File: src/azure-cli/azure/cli/command_modules/appconfig/_params.py (hunk 2 above)
Comment (shenmuxiaosen) [reply src/azure-cli/azure/cli/command_modules/appconfig/_params.py]: We are trying to keep consistent with ```az storage account update``` as below https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/storage/_params.py#L210 As cmk is kind of  new required features for all Azure services, we don't want to invert our own way.

File: src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py (hunk 3 above)
Comment (shenmuxiaosen) [reply src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py]: Due to a backend service behavior changes, those test cases were invalid now. The scenario of these testing code won't be supported any more.

**Comment by qwordy** [comment]
I suggest add test cases for new parameters.

File: src/azure-cli/azure/cli/command_modules/appconfig/_params.py (hunk 1 above)
Comment (shenmuxiaosen) [reply src/azure-cli/azure/cli/command_modules/appconfig/_params.py]: You mean just name the argument "--id"? It looks too vague and maybe we can use "--identity" and explain this should be the client of the managed identity?.

**Comment by shenmuxiaosen** [comment]
Done. 

---
In reply to: [584517166](https://github.com/Azure/azure-cli/pull/12102#issuecomment-584517166) [](ancestors = 584517166)

**Comment by shenmuxiaosen** [comment]
> I suggest add test cases for new parameters.

Done

File: src/azure-cli/azure/cli/command_modules/appconfig/_params.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/appconfig/_params.py]: Sorry. I should make my advice clear. I mean usually ID is upper case.

File: src/azure-cli/azure/cli/command_modules/appconfig/_params.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/appconfig/_params.py]: Sorry. I should make my advice clear. I mean usually URI is upper case.

Comment (shenmuxiaosen) [reply src/azure-cli/azure/cli/command_modules/appconfig/_params.py]: fixed.

File: src/azure-cli/azure/cli/command_modules/appconfig/_params.py (hunk 1 above)
Comment (shenmuxiaosen) [reply src/azure-cli/azure/cli/command_modules/appconfig/_params.py]: fixed.

**Comment by qwordy** [comment]
hi, could you help resolve the conflicting files? thanks.

**Comment by shenmuxiaosen** [comment]
> hi, could you help resolve the conflicting files? thanks.

Done

**Pull request merged**